pkg/controller/flink/config.go

```diff
@@ -124,8 +129,21 @@ func renderFlinkConfig(app *v1beta1.FlinkApplication) (string, error) {
 	(*config)["query.server.port"] = getQueryPort(app)
 	(*config)["blob.server.port"] = getBlobPort(app)
 	(*config)["metrics.internal.query-service.port"] = getInternalMetricsQueryPort(app)
-	(*config)["jobmanager.heap.size"] = getJobManagerHeapMemory(app)
-	(*config)["taskmanager.heap.size"] = getTaskManagerHeapMemory(app)
+
+	appVersion, err := version.NewVersion(getFlinkVersion(app))
+	v10, _ := version.NewVersion("1.10")
+	v11, _ := version.NewVersion("1.11")
+
+	if err != nil || appVersion == nil || appVersion.LessThan(v10) {
+		(*config)["jobmanager.heap.size"] = getJobManagerMemory(app)
```

Comment:
I think we need to be careful about conflating these configs (heap.size and process.size), since they control different things and will lead to different behavior when upgrading.

For existing users (who are on Flink <= 1.10 and using offHeapMemoryFraction), the behavior should stay the same; i.e., we shouldn't move to configuring the process size—we should only do that if they specify the new config.

Reply:
Ok I kept the same configuration for versions <= 1.10, thanks!